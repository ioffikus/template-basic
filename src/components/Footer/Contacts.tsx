import React from 'react';

interface IProps {
  ui: Alicanto.Models.UI;
}

interface IState {}

class Contacts extends React.Component<IProps, IState> {
  render() {
    const contactUI = this.props.ui.common.footerContacts.contacts;
    const zip = contactUI.zip.i18n.defaultText;
    const address = contactUI.address.i18n.defaultText;
    const city = contactUI.city.i18n.defaultText;
    const country = contactUI.country.i18n.defaultText;
    const phoneText = contactUI.phone.i18n.defaultText;
    const phoneLink = contactUI.phone.settings.link;
    return (
      <div className="root">
        <style jsx>{`
          a {
            color: inherit;
          }
          a:hover {
            text-decoration: none;
          }
        `}</style>
        {`${zip} ${address}, ${city}, ${country}, `}
        &nbsp;
        <a href={`tel:${phoneLink}`}>{phoneText}</a>
      </div>
    );
  }
}

export default Contacts;
