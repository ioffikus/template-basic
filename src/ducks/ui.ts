import { createAction, createReducer } from 'redux-act';

export const REDUCER = 'UI';
const NS = `${REDUCER}__`;

export interface IUIState extends Alicanto.Models.UI, Store.IReduxCommonState {}

export const initialState: IUIState = {
  isRequestError: false,
  isRequesting: false,
};

const reducer = createReducer<IUIState>({}, initialState);

export const setUI = createAction<Alicanto.Models.UI>(`${NS}SET_SUCCESS`);
reducer.on(setUI, (state, ui) => ({
  ...state,
  ...ui,
  isRequesting: false,
  isRequestError: false,
}));

const setError = createAction(`${NS}SET_ERROR`);
reducer.on(setError, state => ({
  ...initialState,
  isRequestError: true,
}));

const setStart = createAction(`${NS}SET_START`);
reducer.on(setStart, state => ({
  ...state,
  isRequesting: true,
}));

export default reducer;
