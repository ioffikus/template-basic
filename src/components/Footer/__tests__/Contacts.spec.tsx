import React from 'react';
import { shallow } from 'enzyme';
import ui from 'src/store/__mocks__/uiStore';
import Contacts from '../Contacts';

describe(null, () => {
  it('render() correctly', () => {
    const wrapper = shallow(<Contacts ui={ui} />);
    expect(wrapper).toMatchSnapshot();
  });
});
