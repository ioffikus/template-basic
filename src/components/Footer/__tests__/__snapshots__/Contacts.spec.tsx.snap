// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render() correctly 1`] = `
<div
  className="jsx-1404201388 root"
>
  <JSXStyle
    css="a.jsx-1404201388{color:inherit;}a.jsx-1404201388:hover{text-decoration:none;}"
    styleId="1404201388"
  />
  2025 Hamilton Avenue, San Jose, California 95125, USA, 
  Â 
  <a
    className="jsx-1404201388"
    href="tel:+74083767400"
  >
    (408) 376-7400
  </a>
</div>
`;
